src/crepe/api/AI.h

```diff
@@ -0,0 +1,120 @@
+#pragma once
+
+#include "Component.h"
+#include "types.h"
+
+namespace crepe {
+
+/**
+ * \brief The AI component is used to control the movement of an entity using AI.
+ *
+ * The AI component can be used to control the movement of an entity. The AI component can be used
+ * to implement different behaviors such as seeking, fleeing, arriving, and path following.
+ */
+class AI : public Component {
+public:
+	//! The different types of behaviors that can be used
+	enum BehaviorType {
```

Comment:
Volgensmij zijn dit masks.
Waarvoor dient NONE?

Kun je misschien iets toevoegen van msk of mask zodat dit duidelijk is?

Reply:
reageer als je niet wijzigd maar wel resolved